doc/developer-notes.md

```diff
@@ -380,6 +376,17 @@ C++ data structures
   - *Rationale*: Easier to understand what is happening, thus easier to spot mistakes, even for those
   that are not language lawyers
 
+- Initialize all non-static class members where they are defined
+
+  - *Rationale*: Initializing the members in the declaration makes it easy to spot uninitialized ones,
+  and avoids accidentally reading uninitialized memory
+
+```cpp
+struct A {
```

Comment:
You're talking about class members above, but using a struct here. I think there can be good (performance) arguments not to have an auto-generated constructor for a POD struct, so might be better to use a class for illustration.

Reply:
My opinion is that such cases should just be justified with a comment:
```cpp
int a; // Not initialized for performance reasons.

Reply:
Anyway, fixed...